File: src/test/java/liqp/nodes/RecursivePropertyContainer.java
package liqp.nodes;

import liqp.lookup.HasProperties;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class RecursivePropertyContainer implements HasProperties {

  @Nullable
  @Override
  public Object getProperty(@NotNull String propertyName) {
    if (propertyName.equalsIgnoreCase("title")) {
      return "Lord of the Grapes";
    } else {
      return new RecursivePropertyContainer();
    }
  }
}

File: liqp-core/src/test/kotlin/liqp/nodes/RecursivePropertyContainer.kt
package liqp.nodes

import lang.suppress.Suppressions.UNCHECKED_CAST
import liqp.HasProperties

class RecursivePropertyContainer : HasProperties {
  @Suppress(UNCHECKED_CAST)
  override fun <T : Any> getValue(propName: String): T? {
    return if (propName.equals("title", ignoreCase = true)) {
      "Lord of the Grapes" as T?
    } else {
      RecursivePropertyContainer() as T?
    }
  }
}
